### PR TITLE
ci: showcase native check

### PR DESCRIPTION
**Pull request by burkedavison**

### CODE REVIEW

**Comment by burkedavison** [comment]
Expected native image failure due to missing `fake_cert.pem` test resource:
```
Failures (2):
  JUnit Vintage:ITGdch:testCreateClient_withGdchCredentialAndNoAudience_defaultsToEndpointBasedAudience
    MethodSource [className = 'com.google.showcase.v1beta1.it.ITGdch', methodName = 'testCreateClient_withGdchCredentialAndNoAudience_defaultsToEndpointBasedAudience', methodParameterTypes = '']
    => java.lang.NullPointerException
       com.google.showcase.v1beta1.it.ITGdch.prepareCredentials(ITGdch.java:95)
       com.google.showcase.v1beta1.it.ITGdch.setup(ITGdch.java:78)
       java.base@17.0.7/java.lang.reflect.Method.invoke(Method.java:568)
       org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
       org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
       org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
       org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
       org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
       org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
       org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
       [...]
  JUnit Vintage:ITGdch:testCreateClient_withGdchCredentialWithValidAudience_usesCredentialWithPassedAudience
    MethodSource [className = 'com.google.showcase.v1beta1.it.ITGdch', methodName = 'testCreateClient_withGdchCredentialWithValidAudience_usesCredentialWithPassedAudience', methodParameterTypes = '']
    => java.lang.NullPointerException
       com.google.showcase.v1beta1.it.ITGdch.prepareCredentials(ITGdch.java:95)
       com.google.showcase.v1beta1.it.ITGdch.setup(ITGdch.java:78)
       java.base@17.0.7/java.lang.reflect.Method.invoke(Method.java:568)
       org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
       org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
       org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
       org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
       org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
       org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
       org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
       [...]
```

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -208,6 +208,33 @@ jobs:
...
+          sudo chown -R ${USER} /usr/src/
+          curl --location https://github.com/googleapis/gapic-showcase/releases/download/v${SHOWCASE_VERSION}/gapic-showcase-${SHOWCASE_VERSION}-linux-amd64.tar.gz --output /usr/src/showcase/showcase-${SHOWCASE_VERSION}-linux-amd64.tar.gz
+          cd /usr/src/showcase/
+          tar -xf showcase-*
+          ./gapic-showcase run &
+          cd -
+      - name: Build native image
+        working-directory: showcase
+        run: mvn test -Pnative,-showcase -ntp -B
+
```
Comment (mpeddada1) [inline .github/workflows/ci.yaml:237]: @burkedavison I think we already run showcase tests with native image compilation in our Kokoro presubmits so I wonder if this is still needed? : https://github.com/googleapis/sdk-platform-java/blob/a377aab5149aedb55426a77335d38923a256ab28/.kokoro/presubmit/downstream-build.sh#L78

Comment (mpeddada1) [reply .github/workflows/ci.yaml:237]: Hm but it also looks like the `Kokoro - Test: Downstream Java 17 GraalVM Native Image` job hasn't been running in the most recent PRs.

Comment (burkedavison) [reply .github/workflows/ci.yaml:237]: @mpeddada1 : I made the suggested fix, but the Kokoro builds aren't triggering.

Perhaps we can try to get them reworking separately, and decide then which of the two we want to keep?

File: showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITGdch.java [hunk 2]
```diff
@@ -76,37 +79,44 @@ public class ITGdch {
...
+  private void prepareCredentials() throws IOException {
+    // Copy file so it can be referenced by Path even in native-image builds
+    File caCertFile = tempFolder.newFile(CA_CERT_FILENAME);
+    try (InputStream inputStream = getClass().getResourceAsStream(CA_CERT_RESOURCE_PATH)) {
+      Truth.assertThat(inputStream).isNotNull();
+      Files.copy(inputStream, caCertFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+    }
+    Truth.assertWithMessage(caCertFile.toPath() + " should exist")
+        .that(caCertFile.exists())
+        .isTrue();
```
Comment (mpeddada1) [inline showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITGdch.java:105]: nit: how about we add a static import for this?

**Review by mpeddada1 — APPROVED** [review]

File: showcase/gapic-showcase/src/test/resources/META-INF/native-image/resource-config.json [hunk 3]
```diff
@@ -1,3 +1,6 @@
 {
-  "resources":[{"pattern": ".*.json"}]
+  "resources": [
+    { "pattern": ".*.json" },
+    { "pattern": ".*.pem" }
```
Comment (blakeli0) [inline showcase/gapic-showcase/src/test/resources/META-INF/native-image/resource-config.json:4]: If I'm understanding correctly, only this change is required for native image testing? Are the changes in ITGdch required or refactoring?

Comment (burkedavison) [reply showcase/gapic-showcase/src/test/resources/META-INF/native-image/resource-config.json:4]: The `ITGdch` changes are required. The previous logic created a Path reference to a bundled resource. For normal testing, this path is resolved to the `/target/classes` directory -- but during native image testing, we can't create a Path reference to a resource located inside the native binary. Instead, we have to copy it out to location on the file system and create a path to that copy.

Comment (blakeli0) [reply showcase/gapic-showcase/src/test/resources/META-INF/native-image/resource-config.json:4]: I see, thanks!

**Review by blakeli0 — APPROVED** [review]

**Pull request merged**